src/containers/GuestInfoContainer.tsx

```diff
@@ -0,0 +1,176 @@
+import * as Yup from 'yup';
+import { useCallback, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import { PersonalInfoRequest } from '../api/PersonalInfoRequest';
+import Logger from '../utils/logger';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { LoadingButton } from '@mui/lab';
+import { Box, Card, Checkbox, Grid, Link, Stack, Typography } from '@mui/material';
+import { useAppDispatch, useAppState } from '../store';
+import { FormProvider, RHFTextField } from '../components/hook-form';
+import type { PersonalInfo } from '../store/types';
+import { DateTime } from 'luxon';
+import { MessageBox } from '../components/MessageBox';
+import { RHFDatePicker } from '../components/hook-form/RHFDatePicker';
+import { regexp } from '@windingtree/org.id-utils';
+
+const logger = Logger('GuestInfoContainer');
+
+const defaultValues = {
+  firstname: '',
+  lastname: '',
+  birthdate: null,
+  email: '',
+  phone: ''
+};
+
+export const GuestInfoContainer = () => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const { checkout } = useAppState();
+
+  const NewUserSchema = Yup.object().shape({
+    firstname: Yup.string()
+      .trim()
+      .matches(/^[a-z ,.'-]+$/i, 'Only a-z, A-Z chars')
+      .required('First name is required'),
+    lastname: Yup.string()
+      .trim()
+      .matches(/^[a-z ,.'-]+$/i, 'Only a-z, A-Z chars')
+      .required('Last name is required'),
+    email: Yup.string()
+      .trim()
+      .required('Email is required')
+      .matches(regexp.email, 'Incorrect email')
+      .email(),
+    phone: Yup.string()
+      .trim()
+      .matches(regexp.phone, 'Incorrect phone number')
+      .required('Phone number is required'),
+    birthdate: Yup.string()
+      .matches(regexp.isoDate, 'Incorrect date')
+      .required('Birthdate is required')
+  });
+
+  const [privacy, setPrivacy] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<undefined | string>();
+
+  const methods = useForm<PersonalInfo>({
+    resolver: yupResolver(NewUserSchema),
+    defaultValues
+  });
+
+  const {
+    watch,
+    handleSubmit,
+    formState: { isSubmitting }
+  } = methods;
+
+  const values = watch();
+
+  const onSubmit = useCallback(async () => {
```

Comment:
Instead of using `watch`, you can use the received `data` as prop in the `onSubmit` function to send the data to the BE. 

This is the default way `react-hook-form` recommends doing it, although `watch` also works.